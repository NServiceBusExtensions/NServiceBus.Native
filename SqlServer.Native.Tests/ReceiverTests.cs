using System;
using System.Collections.Generic;
using System.Text;
using ObjectApproval;
using SqlServer.Native;
using Xunit;

public class ReceiverTests
{
    static DateTime dateTime = new DateTime(2000, 1, 1, 1, 1, 1, DateTimeKind.Utc);

    static ReceiverTests()
    {
        DbSetup.Setup();
    }

    [Fact]
    public void ReceiveSingle()
    {
        SqlHelpers.Drop(Connection.ConnectionString, "SendTests").Await();
        QueueCreator.Create(Connection.ConnectionString, "SendTests").Await();
        var sender = new Sender("SendTests");

        var message = BuildMessage("00000000-0000-0000-0000-000000000001");
        sender.Send(Connection.ConnectionString, message).Await();
        var receiver = new Receiver("SendTests");
        var received = receiver.Receive(Connection.ConnectionString).Result;
        ObjectApprover.VerifyWithJson(received);
    }

    [Fact]
    public void ReceiveBatch()
    {
        SqlHelpers.Drop(Connection.ConnectionString, "SendTests").Await();
        QueueCreator.Create(Connection.ConnectionString, "SendTests").Await();
        var sender = new Sender("SendTests");

        sender.Send(
            Connection.ConnectionString,
            new List<Message>
            {
                BuildMessage("00000000-0000-0000-0000-000000000001"),
                BuildMessage("00000000-0000-0000-0000-000000000002")
            }).Await();

        var receiver = new Receiver("SendTests");
        var messages = new List<Message>();
        receiver.Receive(
            connection: Connection.ConnectionString,
            batchSize: 10,
            action: message => { messages.Add(message); })
            .Await();
        ObjectApprover.VerifyWithJson(messages);
    }

    static Message BuildMessage(string guid)
    {
        return new Message(new Guid(guid), "theCorrelationId", "theReplyToAddress", dateTime, "headers", Encoding.UTF8.GetBytes("{}"));
    }
}