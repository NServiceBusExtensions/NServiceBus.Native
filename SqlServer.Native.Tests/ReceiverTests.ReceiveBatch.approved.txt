[
  {
    "Id": "00000000-0000-0000-0000-000000000001",
    "RowVersion": 1,
    "CorrelationId": "theCorrelationId",
    "ReplyToAddress": "theReplyToAddress",
    "Expires": "2000-01-01T01:01:01",
    "Headers": "headers",
    "Body": "e30="
  },
  {
    "Id": "00000000-0000-0000-0000-000000000002",
    "RowVersion": 2,
    "CorrelationId": "theCorrelationId",
    "ReplyToAddress": "theReplyToAddress",
    "Expires": "2000-01-01T01:01:01",
    "Headers": "headers",
    "Body": "e30="
  },
  {
    "Id": "00000000-0000-0000-0000-000000000003",
    "RowVersion": 3,
    "CorrelationId": "theCorrelationId",
    "ReplyToAddress": "theReplyToAddress",
    "Expires": "2000-01-01T01:01:01",
    "Headers": "headers",
    "Body": "e30="
  }
]